sdk/ml/azure-ai-ml/azure/ai/ml/_utils/_asset_utils.py

```diff
@@ -652,6 +652,32 @@ def _create_or_update_autoincrement(
     return result
 
 
+def _get_next_version_from_container(
+    name: str,
+    container_operation: Any,
+    resource_group_name: str,
+    workspace_name: str,
+    registry_name: str,
+    **kwargs,
+) -> str:
+    try:
+        container = container_operation.get(
+            name=name,
+            resource_group_name=resource_group_name,
+            registry_name=registry_name,
+            **kwargs,
+        ) if registry_name else container_operation.get(
+            name=name,
+            resource_group_name=resource_group_name,
+            workspace_name=workspace_name,
+            **kwargs,
+        )
+        version = container.properties.next_version
+
+    except ResourceNotFoundError:
```

Comment:
In any other exception, let's say timeout or something else, then we don't catch and it throws? Do you want it to hard fail or change this to be a best effort next version?

Reply:
We are only interested on this exception because if it does not exist then we want to use version "1", e.g. create new. Other exceptions will be thrown, will not be handled...